navigation/tracking/scripts/pdaf.py

```diff
@@ -0,0 +1,250 @@
+import numpy as np
+
+"""
+Single object tracking
+
+Sub tasks: 
+
+    Use a kalam gain so that the filter is numerically stable. 
+    SINGULAR MATRICIES
+
+    TEST
+        - Visualize position estimates with clutter.
+        - Visualize velocity estimates with vectors. 
+        - Visualize validation gate. 
+        - Does the frequency of "zero detections within gate" correspond with p of detection? 
+
+    Observations: 
+        P pri and post are not symmetrical.
+        P post grows to values over 1000.
+"""
+
+
+class PDAF:
+    def __init__(self):
```

Comment:
To keep the PDAF open to tweaking when implemented as part of the ROS stack, the hyperparameters should be added as input arguments here. If there ends up being too many (around 7 and up) you can make a wrapper object for the config.